src/functions.jl

```diff
@@ -1,8 +1,11 @@
 # This file is a part of InverseFunctions.jl, licensed under the MIT License (MIT).
 
 """
-    square(x)
+    square(x::Real)
 
 Inverse of `sqrt(x)` for non-negative `x`.
 """
-square(x) = x^2
+function square(x::Real)
```

Comment:
I guess we should keep this definition for non-`Real`s, such as `Complex`:
```julia
julia> sqrt(-2 + 0*im)^2
-2.0000000000000004 + 0.0im
```
And it would also cover the matrix case. I wonder what the correct domain check would be there. This is not a problem with this approach specifically though but also for the `invertibleat` suggestion.

```suggestion
square(x) = x^2
function square(x::Real)
```